src/Kernel/client/kernel.ts

```diff
@@ -31,20 +31,43 @@ class Kernel {
                 console.log("my message received", message);
 
                 //create buttons as DOM objects in order to attach unique event handlers
-                let state_div = message.content.state.div_id;
+                let state: DisplayableState = message.content.state;
+                let state_div = state.div_id;
                 if (state_div != null) {
                     let div = document.getElementById(state_div);
                     if (div != null) {
+                        let button1 = document.createElement("button");
+                        let graph = document.createElement("canvas");
+                        button1.appendChild(document.createTextNode("Show Basis States vs Amplitude Squared"));
+                        button1.addEventListener("click", event => {
+                            createBarChart(graph, state);
+                            div.appendChild(graph);
+                        });
+                        div.appendChild(button1);
+                        
+
                         let button = document.createElement("button");
-                        let p = document.createElement("p");
-                        button.appendChild(document.createTextNode("Number of Qubits"));
-                        button.addEventListener("click", event => {
-                            p.innerHTML = message.content.state.n_qubits;
+                        let realImagGraph = document.createElement("canvas");
+                        button.appendChild(document.createTextNode("Show Basis States vs Real,Imag"));
+                        button.addEventListener("click", event => { 
+                            createBarChartRealImagOption(realImagGraph, state);
+                            div.appendChild(realImagGraph);
                         });
                         div.appendChild(button);
-                        div.appendChild(p);
+
+                        let button2 = document.createElement("button");
```

Comment:
Similarly here and at line 49, more descriptive names could help differentiate the three buttons.